osquery/registry/registry_factory.cpp

```diff
@@ -22,7 +22,7 @@
 
 namespace osquery {
 
-HIDDEN_FLAG(bool, registry_exceptions, false, "Allow plugin exceptions");
+HIDDEN_FLAG(bool, registry_exceptions, true, "Allow plugin exceptions");
```

Comment:
I am confused. 

Your PR says:
> to true, so that all tests on tables will fail if an exception

But the flag's description is the inverse.

Reply:
@directionless The description means that when true, you allow the rest of the code to see plugin (or table) exceptions, and that they will not be caught at a lower level and hidden.

The exceptions are always caught at the lower level by a fixed try catch, the difference is if the exception is re-thrown so that the higher levels see it or not. The default is that it's not re-thrown.

Reply:
Would `true` = "allow plugin to _throw_ exceptions (exceptions will not be caught and suppressed)" be a clearer explanation string?

Reply:
It would be slightly incorrect; throwing and catching are two separate things, what that flags does is just not catching or suppressing (if set to true) the exceptions, which is always thrown.

Reply:
> you allow the rest of the code to see plugin (or table) exceptions, and that they will not be caught at a lower level and hidden

I think the issue is that "allow" is deeply ambiguous, and depends a bit on your focus. "allow higher code to see the exception" feels much less natural to me than "allow this to happen (eg: ignore)". Honestly, I would have named the flag `ignore_registry_exceptions` and defaulted it to false. 

Anyhow, maybe something like `When true, treat plugin exceptions as fatal. Otherwise suppress them`

Any maybe apart from the words, you're changing the default to not-suppress?

Reply:
> And maybe apart from the words, you're changing the default to not-suppress?

Yes, that was my initial only focus, because to set that flag to true and suppress them is easily done in one place.
For tests you would need to remember to do it everytime and you end up risking suppressing them and not being able to fix them.

Anyway it's a hidden flag, I would say that we should be able to freely rename that.

Reply:
I agree with being free to rename hidden flags. Is `init.cpp` skipped when running tests? I assume so, otherwise this change would have no effect.

Reply:
Maybe this?
```suggestion
HIDDEN_FLAG(bool, registry_exceptions, true, "Disable suppression of plugin exceptions");
```

Reply:
I can go with Zach's suggestion. Or we can rename the flag to remove the weird double negative.